packages/user/src/tokens.ts

```diff
@@ -0,0 +1,352 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+
+import { AccordionPanel, Menu } from '@lumino/widgets';
+import { ISignal } from '@lumino/signaling';
+import { Token } from '@lumino/coreutils';
+
+/**
+ * An ID to track the user on StateDB.
+ */
+export const USER = '@jupyterlab/user:userDB';
+
+/**
+ * The user token.
+ *
+ * NOTE: Requirer this token in your extension to access the
+ * current connected user information.
+ */
+export const ICurrentUser = new Token<IUser>('@jupyterlab/user:ICurrentUser');
+
+/**
+ * The user menu token.
+ *
+ * NOTE: Requirer this token in your extension to access the user menu
+ * (top-right menu in JupyterLab's interface).
+ */
+export const IUserMenu = new Token<Menu>('@jupyterlab/user:IUserMenu');
```

Comment:
You should create an interface IUserMenu that describe the API of the token and not using a default type directly (expose only what is needed).